velox/functions/prestosql/GeometryFunctions.h

```diff
@@ -542,6 +557,58 @@ struct StYFunction {
   }
 };
 
+template <typename T>
+struct StXMinFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE Status
+  call(out_type<double>& result, const arg_type<Geometry>& geometry) {
+    return computeFromEnvelope(
+        result, geometry, [](const geos::geom::Envelope* env) {
+          return env->getMinX();
+        });
+  }
+};
+
+template <typename T>
+struct StYMinFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE Status
+  call(out_type<double>& result, const arg_type<Geometry>& geometry) {
+    return computeFromEnvelope(
+        result, geometry, [](const geos::geom::Envelope* env) {
+          return env->getMinY();
+        });
```

Comment:
I actually find it easier to read like 
```suggestion
    // maybe std::unique_ptr<...::Envelope> instead
    geos::geom::Envelope envelope = geospatial::readEnvelope(geometry); 
    return envelope->getMinY();
```

We'll use envelopes in other computations later, so this gives us more optionality.